packages/components/src/custom-select-control/index.js

```diff
@@ -12,7 +12,15 @@ import { __, sprintf } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
-import { Button, VisuallyHidden } from '../';
+import { Button, Popover, VisuallyHidden } from '../';
+
+const OptionList = ( { isOpen, children } ) => {
```

Comment:
Instead of a custom implementation like that, maybe `DropdownMenu` is a component suited for this use-case (with all the focus management...)